frontend/projects/ui/src/app/modals/generic-input/generic-input.component.ts

```diff
@@ -37,13 +42,27 @@ export class GenericInputComponent {
   }
 
   toggleMask() {
-    this.unmasked = !this.unmasked
+    this.masked = !this.masked
   }
 
   cancel() {
     this.modalCtrl.dismiss()
   }
 
+  transformInput(newValue: string) {
+    let i = 0
+    this.value = newValue
+      .split('')
+      .map(x => {
```

Comment:
`x => x === '●' ? this.value[i++] : x` for shorts :) Why does it return next char from old value for each '●' in new value though?

Reply:
This is so we can accurately persist the actual text value when it is displaying as the masked value as the user is typing. Ie. `newValue` could be `maskedValue` (just dots)  or `value` (plaintext) depending if input is toggled to masked or not.